### PR TITLE
feat: add encoding ngsderive module

### PR DESCRIPTION
**Pull request by a-frantz**

### CODE REVIEW

File: workflows/qc/quality-check-standard.wdl [hunk 1]
```diff
@@ -91,6 +91,9 @@ workflow quality_check {
         call samtools.subsample as samtools_subsample { input: bam=validate_bam.validated_bam, max_retries=max_retries }
         call picard.bam_to_fastq { input: bam=samtools_subsample.sampled_bam, max_retries=max_retries }
         call fq.fqlint { input: read1=bam_to_fastq.read1, read2=bam_to_fastq.read2, max_retries=max_retries }
+        String prefix = basename(bam_to_fastq.read1, "_R1.fastq")
+        Array[File] fastqs = [bam_to_fastq.read1, bam_to_fastq.read2]
+        call ngsderive.encoding as ngsderive_encoding { input: fastqs=fastqs, prefix=prefix, max_retries=max_retries }
```
Comment (adthrasher) [inline workflows/qc/quality-check-standard.wdl:96]: Should this be passed to fastq_screen?

**Review by adthrasher — APPROVED** [review]

File: workflows/qc/quality-check-standard.wdl (hunk 1 above)
Comment (a-frantz) [reply workflows/qc/quality-check-standard.wdl:96]: Yes this was just an initial commit to test the encoding itself before adding the encoding parsing to fastq_screen.

**Review by adthrasher — APPROVED** [review]

File: tools/ngsderive.wdl [hunk 2]
```diff
@@ -105,8 +105,66 @@ task read_length {
     }
 
     output {
-        String computed_read_length = read_string(out_file)
```
[the remark this answers is not in the document]
Comment (a-frantz) [reply tools/ngsderive.wdl:108]: This line as-is returns the header of the file. It doesn't parse out the value, and isn't used anywhere else.

File: docker/ngsderive/1.1.0/Dockerfile [hunk 3]
```diff
@@ -7,6 +7,6 @@ RUN apt-get update && \
 
 RUN conda create -n ngsderive python==3.8
 ENV PATH /opt/conda/envs/ngsderive/bin/:$PATH
-RUN pip install --no-cache-dir ngsderive==1.1.0
+RUN pip install --no-cache-dir ngsderive==1.3.0
```
[the remark this answers is not in the document]
Comment (a-frantz) [reply docker/ngsderive/1.1.0/Dockerfile:10]: Yes this is a bit confusing... But it's just a coincidence that the versions matched up previously. I think the dockerfiles should be maintaining their own versioning, as opposed to deriving their version from the underlying tool.

File: workflows/qc/quality-check-standard.wdl [hunk 4]
```diff
@@ -106,8 +109,11 @@ workflow quality_check {
     }
     if (experiment == "RNA-Seq") {
         File gencode_gtf_defined = select_first([gencode_gtf, "No GTF"])
+        Array[File] bam_as_array = [validate_bam.validated_bam]
```
[the remark this answers is not in the document]
Comment (a-frantz) [reply workflows/qc/quality-check-standard.wdl:112]: Yes if WDL allows it. But I don't think WDL likes inlining things. It tends to prefer explicit variable declarations. I can check if the spec allows inlining here.

File: docker/ngsderive/1.1.0/Dockerfile (hunk 3 above)
[the remark this answers is not in the document]
Comment (adthrasher) [reply docker/ngsderive/1.1.0/Dockerfile:10]: I think that's fine as long as we're consistent in how we're versioning them.

File: workflows/qc/quality-check-standard.wdl (hunk 4 above)
[the remark this answers is not in the document]
Comment (adthrasher) [reply workflows/qc/quality-check-standard.wdl:112]: It appears to work as I've used it in XenoCP (https://github.com/stjude/XenoCP/blob/b6101250610df1920594b7c06b069f0bd597c694/wdl/workflows/xenocp.wdl#L77). Let me know what the WDL spec allows.

[the remark this answers is not in the document]
Comment (a-frantz) [reply workflows/qc/quality-check-standard.wdl:112]: You're right, it's allowed. Updating now

**Pull request merged**